projects/rocprofiler-sdk/source/lib/output/stream_info.hpp

```diff
@@ -134,6 +134,21 @@ struct tool_buffer_tracing_hip_api_ext_record_t : rocprofiler_buffer_tracing_hip
     rocprofiler_stream_id_t stream_id = {};
 };
 
+struct tool_buffer_tracing_kfd_record_t
+{
+    rocprofiler_buffer_tracing_kind_t kind;
+    union {
+        rocprofiler_buffer_tracing_kfd_event_page_migrate_record_t   event_page_migrate_record;
+        rocprofiler_buffer_tracing_kfd_event_page_fault_record_t     event_page_fault_record;
+        rocprofiler_buffer_tracing_kfd_event_queue_record_t          event_queue_record;
+        rocprofiler_buffer_tracing_kfd_event_unmap_from_gpu_record_t event_unmap_from_gpu_record;
+        rocprofiler_buffer_tracing_kfd_event_dropped_events_record_t event_dropped_events_record;
+        rocprofiler_buffer_tracing_kfd_page_migrate_record_t         page_migrate_record;
+        rocprofiler_buffer_tracing_kfd_page_fault_record_t           page_fault_record;
+        rocprofiler_buffer_tracing_kfd_queue_record_t                queue_record;
+    };
+};
```

Comment:
Maybe just use std::variant here. It would simplify removing the switch statements